krystex/src/main/java/com/flipkart/krystal/krystex/node/Node.java

```diff
@@ -160,18 +162,29 @@ CompletableFuture<NodeResponse> executeRequestCommand(NodeRequestCommand nodeCom
     return resultForRequest;
   }
 
+  public void markRecursive() {
+    this.nodeDefinition = nodeDefinition.toRecursive();
+  }
+
   private CompletableFuture<NodeResponse> handleSkipDependency(
       RequestId requestId, SkipNode skipNode, CompletableFuture<NodeResponse> resultForRequest) {
-    Iterable<ResolverDefinition> pendingResolvers =
-        resolverDefinitionsByInput.values().stream()
-            .flatMap(Collection::stream)
-            .collect(Collectors.toSet());
-    Map<NodeLogicId, ResolverDefinition> uniquePendingResolvers = new LinkedHashMap<>();
-    for (ResolverDefinition pendingResolver : pendingResolvers) {
-      uniquePendingResolvers.putIfAbsent(pendingResolver.resolverNodeLogicId(), pendingResolver);
-    }
-    for (ResolverDefinition resolverDefinition : uniquePendingResolvers.values()) {
-      executeResolver(requestId, resolverDefinition);
+    if (nodeDefinition.isRecursive()) {
+      /*
```

Comment:
should we have the reverse check instead?

Reply:
Done